docs/decisions/0008-application_state_differenciation.rst

```diff
@@ -0,0 +1,176 @@
+
+The re-fetch behavior of ``react-query`` results in page components that aren't dependent on the updated context data to reload.
+
+The re-fetch behavior was occurring in the background of a component while the user is interacting with the page or if they had returned focus back onto the browser while interacting with other tabs or applications.
+
+An example where populating context data from ``react-query`` cache data resulted in unintentional behavior is the ``EnterpriseSubsidiesContext`` containing API data from the query key 'learner-credit-management.budgets', ``isFetchingBudgets`` attribute which has since been fixed.
+
+While the user was interacting with a ``FullscreenModal`` component and had not completed their intended activity within the modal before the global ``staleTime`` had expired, the re-fetch behavior would trigger resulting in a full page re-load, thus closing the modal without saving the users state information (form data etc.).
+
+Decisions
+*********
+
+We will take a more intentional approach when implementing API data fetching using the ``react-query`` in the following ways:
+
+#. Avoid adding API data fetched using ``react-query`` to a context layer
+    * If a context layer does require API data to exist in the context layer, do not use ``react-query`` to retrieve the data
+#. When using ``react-query`` make the call as close to the component that uses the returned response data
+    * This may require mapping a component from the parent component which completes the API call
+#. Leverage the `query key factory <https://github.com/openedx/frontend-app-admin-portal/blob/c67c5e4d8a0328fe75cb9d46791a8b733fad8257/src/components/learner-credit-management/data/constants.js#L67-L77>`_ setup to simplify boilerplate code when making the `useQuery` calls and `invalidateQueries`
+#. Implement a hook when calling the ``useQuery`` method to avoid duplicate code and easy maintainability. See this `hook <https://github.com/openedx/frontend-app-admin-portal/blob/c67c5e4d8a0328fe75cb9d46791a8b733fad8257/src/components/EnterpriseSubsidiesContext/data/hooks.js#L108-L120>`_ as boilerplate as an example, there are `others to reference in the repo <https://github.com/search?q=repo%3Aopenedx%2Ffrontend-app-admin-portal+%2FuseQuery%5C%28%2F&type=code>`_ as additional guides
+#. Invalidate queries when user interaction results in an API data change to avoid stale data
+    * If the query is not invalidated, the re-fetching of the data is entirely dependent on the length of time set for the ``staleTime``
+
+Consequences
+************
+
+Based on the call you're making, it may be challenging to coalesce all the required attributes to make the appropriate API call.
```

Comment:
[curious] I'm wondering if this bit around `mapStateToProps` is necessary to include? It's not really directly related to keeping server state in query cache vs. server state in context. Regardless of either approach, the coalescing of such metadata from the Redux store would be a consideration, no?

I think the ADR could probably be simplified if these Redux concerns are excluded.

Reply:
Noted and removed 👍🏽